### PR TITLE
BUG: remote_file_list failures

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #348.

`inst.remote_file_list` failed to produce a list of files for some instruments.  This was due to the method it used to choose the expected file preamble.  This has been fixed and cleaned up.

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

```
import pysat     
ivm = pysat.Instrument('cnofs', 'ivm')
ivm.remote_file_list(year=2009, month=1) 
```
should yield 
```
2009-01-01    cnofs_cindi_ivm_500ms_20090101_v01.cdf
2009-01-02    cnofs_cindi_ivm_500ms_20090102_v01.cdf
2009-01-03    cnofs_cindi_ivm_500ms_20090103_v01.cdf
2009-01-04    cnofs_cindi_ivm_500ms_20090104_v01.cdf
2009-01-05    cnofs_cindi_ivm_500ms_20090105_v01.cdf
2009-01-06    cnofs_cindi_ivm_500ms_20090106_v01.cdf
2009-01-07    cnofs_cindi_ivm_500ms_20090107_v01.cdf
2009-01-08    cnofs_cindi_ivm_500ms_20090108_v01.cdf
2009-01-09    cnofs_cindi_ivm_500ms_20090109_v01.cdf
2009-01-10    cnofs_cindi_ivm_500ms_20090110_v01.cdf
2009-01-11    cnofs_cindi_ivm_500ms_20090111_v01.cdf
2009-01-12    cnofs_cindi_ivm_500ms_20090112_v01.cdf
2009-01-13    cnofs_cindi_ivm_500ms_20090113_v01.cdf
2009-01-14    cnofs_cindi_ivm_500ms_20090114_v01.cdf
2009-01-15    cnofs_cindi_ivm_500ms_20090115_v01.cdf
2009-01-16    cnofs_cindi_ivm_500ms_20090116_v01.cdf
2009-01-17    cnofs_cindi_ivm_500ms_20090117_v01.cdf
2009-01-18    cnofs_cindi_ivm_500ms_20090118_v01.cdf
2009-01-19    cnofs_cindi_ivm_500ms_20090119_v01.cdf
2009-01-20    cnofs_cindi_ivm_500ms_20090120_v01.cdf
2009-01-21    cnofs_cindi_ivm_500ms_20090121_v01.cdf
2009-01-22    cnofs_cindi_ivm_500ms_20090122_v01.cdf
2009-01-23    cnofs_cindi_ivm_500ms_20090123_v01.cdf
2009-01-24    cnofs_cindi_ivm_500ms_20090124_v01.cdf
2009-01-25    cnofs_cindi_ivm_500ms_20090125_v01.cdf
2009-01-26    cnofs_cindi_ivm_500ms_20090126_v01.cdf
2009-01-27    cnofs_cindi_ivm_500ms_20090127_v01.cdf
2009-01-28    cnofs_cindi_ivm_500ms_20090128_v01.cdf
2009-01-29    cnofs_cindi_ivm_500ms_20090129_v01.cdf
2009-01-30    cnofs_cindi_ivm_500ms_20090130_v01.cdf
2009-01-31    cnofs_cindi_ivm_500ms_20090131_v01.cdf
dtype: object
```

**Test Configuration**:
* Mac OS 10.14.6
* python 3.7.3

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: pysat/instruments/methods/nasa_cdaweb.py [hunk 1]
```diff
@@ -488,16 +488,16 @@ def list_remote_files(tag, sat_id,
...
             soup = BeautifulSoup(requests.get(temp_url).content, "lxml")
             links = soup.find_all('a', href=True)
             for link in links:
-                if level < n_layers:
-                    # If there is room to go down, look for directories
-                    if link['href'].count('/') == 1:
-                        remote_dirs[level+1].append(link['href'])
+                # If there is room to go down, look for directories
+                if link['href'].count('/') == 1:
+                    remote_dirs[level+1].append(link['href'])
```
Comment (jklenzing) [inline pysat/instruments/methods/nasa_cdaweb.py]: The double-check here should not be necessary.

File: pysat/instruments/methods/nasa_cdaweb.py [hunk 2]
```diff
@@ -525,5 +525,6 @@ def list_remote_files(tag, sat_id,
             mask = mask & (stored_list.index.month == month)
             if day is not None:
                 mask = mask & (stored_list.index.day == day)
+        stored_list = stored_list[mask]
 
-    return stored_list[mask]
+    return stored_list
```
Comment (jklenzing) [inline pysat/instruments/methods/nasa_cdaweb.py]: Since the mask does not always exist, cleaned this up.

**Comment by jklenzing** [comment]
Closed and reopened to trick Travis-CI into rerunning the tests.

File: pysat/instruments/methods/nasa_cdaweb.py [hunk 3]
```diff
@@ -273,7 +273,7 @@ def download(supported_tags, date_array, tag, sat_id,
         if not multi_file_day:
             try:
                 logger.info(' '.join(('Attempting to download file for',
-                                date.strftime('%d %B %Y'))))
+                            date.strftime('%d %B %Y'))))
```
Comment (rstoneback) [inline pysat/instruments/methods/nasa_cdaweb.py]: I don't think some of PEP8 formatting will ever look 'right' to me. So it goes. I'd actually push the lines more to the right to emphasize they are in the inner parenthesis.

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,19 +2,20 @@
...
 - Documentation
   - Fixed description of tag and sat_id behaviour in testing instruments
 - Bug Fix
-   - `_files._attach_files` now checks for an empty file list before appending
-   - Fixed boolean logic in when checking for start and stop dates in `_instrument.download`
-   - Fixed loading of COSMIC atmPrf files
-   - Fixed feedback from COSMIC GPS when data not found on remote server
-   - Fixed deprecation warning for pysat.utils.coords.scale_units
+  - `_files._attach_files` now checks for an empty file list before appending
+  - Fixed boolean logic in when checking for start and stop dates in `_instrument.download`
```
Comment (rstoneback) [inline CHANGELOG.md]: I think this should be, 
Fixed boolean logic when checking for start and stop dates in `_instrument.download`

File: pysat/instruments/methods/nasa_cdaweb.py (hunk 3 above)
Comment (jklenzing) [reply pysat/instruments/methods/nasa_cdaweb.py]: D'oh.  You're right.  These should be to the right of `join`.

File: CHANGELOG.md (hunk 4 above)
Comment (jklenzing) [reply CHANGELOG.md]: fixed

File: pysat/instruments/methods/nasa_cdaweb.py (hunk 3 above)
Comment (jklenzing) [reply pysat/instruments/methods/nasa_cdaweb.py]: fixed.

**Pull request merged**